packages/@aws-cdk/aws-stepfunctions-tasks/lib/sagemaker/base-types.ts

```diff
@@ -638,7 +638,7 @@ export interface TransformResources {
   /**
    * ML compute instance type for the transform job.
    */
-  readonly instanceType: ec2.InstanceType;
+  readonly instanceType: ec2.InstanceType | string;
```

Comment:
Typescript union types will not be compatible with other languages.
This will need to be represented by a single type. The way we do it in the cdk is to create a union like class with APIs like `fromInstanceType` and `fromString` that return that type.

Reply:
I see, I can define a `instanceType` class here to use it as the type of this field, something like this:
```
export class InstanceType {
  public static fromEc2InstanceType(instanceType: ec2.InstanceType) {
    return new InstanceType(`ml.${instanceType}`);
  }

  public static fromString(instanceType: string) {
    return new InstanceType(instanceType);
  }

  constructor(private readonly instanceType: string) {
  }

  public toString(): string {
    return this.instanceType;
  }
}
```

But it won't be backward compatible, e.g. current users need to change their code: `instanceType: ec2.InstanceType.of(ec2.InstanceClass.P3, ec2.InstanceSize.XLARGE2)`. Should we make it backward compatible somehow, or it's fine to do this?

Reply:
Hi @shivlaks, could you take a quick look at the above comments when you get a chance? Thanks!

Reply:
Hi @shivlaks, looks like `yarn compat` is not happy with this change, because it changed the API. So should I make the `InstanceType` class defined in above to extend `ec2. InstanceType`?